lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -5991,6 +5991,12 @@ class PullbackEmitter final : public SILInstructionVisitor<PullbackEmitter> {
   AdjointValue accumulateAdjointsDirect(AdjointValue lhs, AdjointValue rhs,
                                         SILLocation loc);
 
+  /// Given two adjoint values, where one is a concrete value and another
+  /// is an aggregate, accumulate them.
+  AdjointValue accumulateAdjointsDirect(SILValue concrete,
```

Comment:
Note: defining this helper function `accumulateAdjointsDirect` on `PullbackEmitter` has the downside of making it accessible to all `PullbackEmitter` code.

On the other hand, defining the helper as a top-level static function isn't possible without significant refactoring because it needs to call private methods of `PullbackEmitter` (e.g. `accumulateAdjointsDirect`).

How about using a more descriptive name so that this helper doesn't get called accidentally? There are only two callers so a long name doesn't hurt too bad.
```suggestion
  AdjointValue accumulateConcreteAndAggregateAdjointsDirect(SILValue concrete,
```